pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -0,0 +1,71 @@
+package inferredspan
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/serverless/tags"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// tagInferredSpanTagSource is the key to the meta tag
+	// that lets us know whether this span should inherit its tags.
+	// Expected options are "lambda" and "self"
+	tagInferredSpanTagSource = "_inferred_span.tag_source"
+
+	// additional function specific tag keys to ignore
+	functionVersionTagKey = "function_version"
+	coldStartTagKey       = "cold_start"
+)
+
+var functionTagsToIgnore = [...]string{
+	tags.FunctionARNKey,
+	tags.FunctionNameKey,
+	tags.ExecutedVersionKey,
+	tags.EnvKey,
+	tags.VersionKey,
+	tags.ServiceKey,
+	tags.RuntimeKey,
+	tags.MemorySizeKey,
+	tags.ArchitectureKey,
+	functionVersionTagKey,
+	coldStartTagKey,
+}
+
+// Check determines if a span belongs to a managed service or not
+// _inferred_span.tag_source = "self" => managed service span
+// _inferred_span.tag_source = "lambda" or missing => lambda related span
+func Check(span *pb.Span) bool {
+	return strings.Compare(span.Meta[tagInferredSpanTagSource], "self") == 0
+}
+
+// FilterFunctionTags filters out DD tags & function specific tags
+func FilterFunctionTags(input map[string]string) map[string]string {
+
+	if input == nil {
+		return nil
+	}
+
+	output := input
```

Comment:
It looks like this function returns an output map, but because maps are a reference type, I think you are also changing the input map here. To clarify this I would update this function so it doesn't return anything and just modifies the input.

Reply:
That's how I originally wrote this function but @maxday suggested we return a new object in favor of making this a pure function so I created an output map (but didn't deep copy it). I'll update the code so it deep copies the input map.